forage-android/src/main/java/com/joinforage/forage/android/network/data/CapturePaymentRepository.kt

```diff
@@ -66,8 +84,8 @@ internal class CapturePaymentRepository(
     private suspend fun collectPinToCapturePayment(
         paymentRef: String,
         cardToken: String,
-        encryptionKey: String
-
+        encryptionKey: String,
+        responseMonitor: ResponseMonitor
```

Comment:
**Can ignore this comment if we purse the `ForageSDK.kt` approach described in my other comment**

> Should we call end at the entry ForageSDK.kt level and set a boundary around the beginning of the checkBalance method and the time when the client gets a response.

----
to avoid drilling the `responseMonitor` through every request method

could we keep track of `responseMonitor` in a map?

```kotlin
internal object MeasurementStore {
  private val store: HashMap<EventName, ResponseMonitor> = HashMap()

  fun set(key: EventName, instance: ResponseMonitor) {
     store[key] = instance
  }

  fun get(key: EventName): ResponseMonitor? {
    return store[key]
  }
}

customerPerceivedMonitor.start()
MeasurementStore.set(CUSTOMER_PERCEIVED, customerPerceivedMonitor)


val customerPerceivedMonitor = MeasurementStore.get(CUSTOMER_PERCEIVED)
customerPerceivedMonitor.end()
```

@devinmorgan would also like to get your thoughts on alternatives to parameter drilling the responseMonitor.